From: Balint Reczey <balint.reczey@canonical.com>
Date: Sun, 22 Mar 2020 21:37:06 +0100
Subject: units: Don't mount tracefs in LXC

Mounting tracefs fails in unprivileged LXC containers and latest LXD
(3.23) bind mounts tracefs already.

Fixes https://github.com/lxc/lxd/issues/7059

(cherry picked from commit cdc6c95737d26b1746b446dfe662872d593d23c6)
---
 units/sys-kernel-tracing.mount | 1 +
 1 file changed, 1 insertion(+)

diff --git a/units/sys-kernel-tracing.mount b/units/sys-kernel-tracing.mount
index cb42b47..e29b258 100644
--- a/units/sys-kernel-tracing.mount
+++ b/units/sys-kernel-tracing.mount
@@ -12,6 +12,7 @@ Description=Kernel Trace File System
 Documentation=https://www.kernel.org/doc/Documentation/trace/ftrace.txt
 Documentation=https://www.freedesktop.org/wiki/Software/systemd/APIFileSystems
 DefaultDependencies=no
+ConditionVirtualization=!lxc
 ConditionPathExists=/sys/kernel/tracing
 ConditionCapability=CAP_SYS_RAWIO
 Before=sysinit.target
